cmake/compiler/gcc.cmake

```diff
@@ -50,56 +58,76 @@ foreach(file_name include include-fixed)
   list(APPEND NOSTDINC ${_OUTPUT})
 endforeach()
 
-include(${ZEPHYR_BASE}/cmake/gcc-m-cpu.cmake)
+if("${ZEPHYR_TOOLCHAIN_VARIANT}" STREQUAL "zephyr")
```

Comment:
This is incorrect.

We want to append -mthumb for more toolchain's than 'zephyr', for instance 'gccarmemb' would also use this build script code.

This if statement must be removed.

Reply:
Will rearrange the "ifs" ensuring the else part falls here.